packages/react/src/components/table/utils/table-utils.ts

```diff
@@ -32,19 +32,35 @@ export function calculateStickyHeader(
     });
 }
 
+export function calculateStickyFooter(
+    stickyColumns: boolean[],
+    footerCells: HTMLCollectionOf<HTMLTableCellElement>,
+): void {
+    Array.from(footerCells).forEach((footerCell, index) => {
+        footerCell.style.setProperty('bottom', '0px');
+        footerCell.style.setProperty('z-index', stickyColumns[index] ? '5' : '4');
+    });
+}
+
 export function calculateStickyPosition(
     stickyColumns: boolean[],
     stickyHeader: boolean,
+    stickyFooter: boolean,
     tableRef: React.RefObject<HTMLTableElement>,
 ): void {
     if (tableRef.current === null) {
         return;
     }
     const headerCells = tableRef.current.getElementsByTagName('th');
     const rows = tableRef.current.getElementsByTagName('tr');
+    const footerCells = tableRef.current.getElementsByTagName('tfoot')[0]?.getElementsByTagName('td');
```

Comment:
We should stop using `getElementsByTagName` and use `querySelector` or `querySelectorAll` instead.

The reason is getElementsByTagName and ClassName are LIVE
> The returned list is live, which means it updates itself with the DOM tree automatically. Therefore, there is no need to call Element.getElementsByTagName() with the same element and arguments repeatedly if the DOM changes in between calls.
https://developer.mozilla.org/en-US/docs/Web/API/Element/getElementsByTagName.

Exemple what that mean and why this is harder to reason and could lead to unexpected results.

```
const oldElements = document.getElementsByTagName("div");
oldElements.length // 3
document.body.appendChild(document.createElement('div'));
oldElements.length // 4

const oldElements = document.querySelectorAll("div");
oldElements.length // 3
document.body.appendChild(document.createElement('div'));
oldElements.length // 3
```

My rule of thumb is to never use them unless we do really need this behavior.

Also querySelector is more modern and faster, and having a unique function for all selectors is more pleasant.

Reply:
done